### PR TITLE
Scripts/Spells: warrior Mortal Strike & Colossus Smash Effect

### PR DESCRIPTION
**Pull request by Hadifaren**
fix spell Mortal Strike healing reduce effect
fix spell Colossus Smash Damage taken effect 
- http://www.wowhead.com/spell=12294/mortal-strike
- http://www.wowhead.com/spell=167105/colossus-smash

**Target branch(es):** 3.3.5/master

- [X] master

**Tests performed:** (Does it build, tested in-game, etc.)
compiled + tested in game, works.

### CODE REVIEW

**Comment by Keader** [comment]
Are you sure that this spell does not hit objects?
If can hit objects, you need put a check in ```GetHitUnit()```

**Comment by Hadifaren** [comment]
check like this?

`Unit* target = GetHitUnit()`

**Comment by Keader** [comment]
y, like:
```cpp
if (Unit* target = GetHitUnit())
    GetCaster()->CastSpell(target, SPELL_WARRIOR_MORTAL_WOUNDS, true);
```

**Comment by tkrokli** [comment]
I think you should write the spell name:**`spell_warr_mortal_strike`** because the enum name is `SPELL_WARRIOR_MORTAL_STRIKE` as well as following the normal standards for spell script names.

See for example the `void AddSC_warrior_spell_scripts()` list at the bottom of `spell_warrior.cpp` for more examples:
```cpp
     new spell_warr_charge_effect();
     new spell_warr_concussion_blow();
     new spell_warr_heroic_leap();
     new spell_warr_intimidating_shout();
     new spell_warr_last_stand();
     new spell_warr_rallying_cry();
     new spell_warr_shattering_throw();
     new spell_warr_storm_bolt();
     new spell_warr_sudden_death();
     new spell_warr_sweeping_strikes();
     new spell_warr_sword_and_board();
     new spell_warr_victory_rush();
```

I have also seen it written `spell_warr_mortal_strike` other places than in TrinityCore, but the important part is to keep the same codestyle as in the other spell names.

File: sql/updates/world/master/8888_88_88_88_world.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM `spell_script_names` WHERE `spell_id`=167105 AND `ScriptName`='spell_warr_colossus_smash';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES (167105, 'spell_warr_colossus_smash');
```
Comment (tkrokli) [inline sql/updates/world/master/8888_88_88_88_world.sql]: Please put the content of this file into your already existing SQL file `9999_99_99_99_world.sql`. If you want to add more SQL commands for the script, use the same SQL file as long as it is used for the `world` database.

Comment (Hadifaren) [inline sql/updates/world/master/8888_88_88_88_world.sql]: sorry tkrokli i thinked my next commit will not be added here so i added another sql file
^_^

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+DELETE FROM `spell_script_names` WHERE `spell_id`=12294 AND `ScriptName`='spell_warr_mortalstrike';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES (12294, 'spell_warr_mortalstrike');
+
+DELETE FROM `spell_script_names` WHERE `spell_id`=167105 AND `ScriptName`='spell_warr_colossus_smash';
```
Comment (tkrokli) [inline sql/updates/world/master/9999_99_99_99_world.sql]: Use this format for your `DELETE FROM` commands:
```sql
DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_warr_mortal_strike';
DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_warr_colossus_smash';
```
The logic here is to make sure you get rid of other spell scripts with the same name (if they exist), but with different (wrong) spell IDs.

As a matter of fact, you could simplify the SQL like this:
```sql
DELETE FROM `spell_script_names` WHERE `ScriptName` IN ('spell_warr_mortal_strike','spell_warr_colossus_smash');
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
( 12294, 'spell_warr_mortal_strike'),
(167105, 'spell_warr_colossus_smash');
```
You can of course keep your 2 `DELETE FROM` commands if you like, if you find that easier to view or if you want to add more later. Just keep in mind that you only need 1 `INSERT INTO` command for the same table. :-)

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 3]
```diff
@@ -557,6 +594,41 @@ class spell_warr_last_stand : public SpellScriptLoader
         }
 };
 
+// 12294 - Mortal Strike 7.1.5
+class spell_warr_mortal_strike : public SpellScriptLoader
+{
+public:
+    spell_warr_mortal_strike() : SpellScriptLoader("spell_warr_mortalstrike") { }
```
Comment (tkrokli) [inline src/server/scripts/Spells/spell_warrior.cpp]: `spell_warr_mortal_strike`

Comment (Hadifaren) [inline src/server/scripts/Spells/spell_warrior.cpp]: xD sorry didnt saw

**Comment by Hadifaren** [comment]
i tested mortal strike on 3target and i see that all 3targets got debuff why?

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 4]
```diff
@@ -240,6 +242,41 @@ class spell_warr_charge_effect : public SpellScriptLoader
...
+public:
+    spell_warr_colossus_smash() : SpellScriptLoader("spell_warr_colossus_smash") { }
+
+    class spell_warr_colossus_smash_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warr_colossus_smash_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARRIOR_COLOSSUS_SMASH_EFFECT))
```
Comment (wonder2011) [inline src/server/scripts/Spells/spell_warrior.cpp]: I thing is good idea to use the new ValidateSpellInfo function.

```
return ValidateSpellInfo
({
    SPELL_WARRIOR_COLOSSUS_SMASH_EFFECT
});
```

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 5]
```diff
@@ -557,6 +594,41 @@ class spell_warr_last_stand : public SpellScriptLoader
...
+public:
+    spell_warr_mortal_strike() : SpellScriptLoader("spell_warr_mortal_strike") { }
+
+    class spell_warr_mortal_strike_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warr_mortal_strike_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARRIOR_MORTAL_WOUNDS))
```
Comment (wonder2011) [inline src/server/scripts/Spells/spell_warrior.cpp]: The same here, use ValidateSpellInfo function.

```
return ValidateSpellInfo
({
    SPELL_WARRIOR_MORTAL_WOUNDS
});
```

**Review by wonder2011 — COMMENTED** [review]
Use the ValidateSpellInfo function.

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 6]
```diff
@@ -557,6 +594,41 @@ class spell_warr_last_stand : public SpellScriptLoader
...
+                return false;
+            return true;
+        }
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* target = GetHitUnit())
```
Comment (Seyden) [inline src/server/scripts/Spells/spell_warrior.cpp]: GetHitUnit() can't be null in OnEffectHitTarget hook

Comment (Keader) [inline src/server/scripts/Spells/spell_warrior.cpp]: can, if hit objects

**Comment by wonder2011** [comment]
What about this PR ?

File: src/server/scripts/Spells/spell_warrior.cpp (hunk 6 above)
Comment (Seyden) [inline src/server/scripts/Spells/spell_warrior.cpp]: How can it hit objects if the target type is TARGET_UNIT_TARGET_ENEMY.. tell me Keader ^^

**Comment by Hadifaren** [comment]
if anything is bad someone should say and then with help of people spells can be fixed

**Comment by Aokromes** [comment]
any reason to don't merge this?

**Comment by HannibalRoG** [comment]
@Aokromes Yes, one reason, it seems that the PR must have more than 5 months to be able to be applied 8(

**Comment by Aokromes** [comment]
@Shauren this can be merged?

File: src/server/scripts/Spells/spell_warrior.cpp (hunk 4 above)
Comment (Mihapro) [reply src/server/scripts/Spells/spell_warrior.cpp]: it's meant to be in one line tho

**Pull request merged**